terratag.go

```diff
@@ -21,34 +22,44 @@ import (
 )
 
 type counters struct {
-	totalResources  int
-	taggedResources int
-	totalFiles      int
-	taggedFiles     int
+	totalResources  uint32
+	taggedResources uint32
+	totalFiles      uint32
+	taggedFiles     uint32
 }
 
 var pairRegex = regexp.MustCompile(`^([a-zA-Z][\w-]*)=([\w-]+)$`)
 
+var matchWaitGroup sync.WaitGroup
+
 func (c *counters) Add(other counters) {
-	c.totalResources += other.totalResources
-	c.taggedResources += other.taggedResources
-	c.totalFiles += other.totalFiles
-	c.taggedFiles += other.taggedFiles
+	atomic.AddUint32(&c.totalResources, other.totalResources)
```

Comment:
this may be accessed by multiple goroutines. Incrementing using atomic increment (could have used a mutex instead).